app/views/papers/_index.html.erb

```diff
@@ -52,8 +52,13 @@
         <% end %>
       </small>
       <ul class="nav nav-pills" role="tablist">
+        <% comments_count = paper_reference ? paper_reference.comments.map {|c| c.self_and_descendants.length}.inject(0, :+).to_s : paper.references.map { |r| r.comments.map {|c| c.self_and_descendants.length}.inject(0, :+)}.inject(0, :+).to_s %>
```

Comment:
The cleanest I was able to get this was:

`paper.references.inject(0) { |total, r| total += r.comments.inject(0) {|memo,c| memo += c.self_and_descendants.length}}`